lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -823,24 +823,10 @@ public function getScalarResult()
     public function getOneOrNullResult($hydrationMode = null)
     {
         try {
-            $result = $this->execute(null, $hydrationMode);
+            return $this->getSingleResult($hydrationMode);
         } catch (NoResultException $e) {
             return null;
         }
-
-        if ($this->_hydrationMode !== self::HYDRATE_SINGLE_SCALAR && ! $result) {
-            return null;
```

Comment:
`getSingleResult` throws an exception here instead of returning `null`.

Reply:
@SenseException yes the new code catch the NoResultException and return null same as before.